codegen/cmd/injection-gen/generators/client.go

```diff
@@ -98,8 +98,13 @@ func withClient(ctx {{.contextContext|raw}}, cfg *{{.restConfig|raw}}) context.C
 func Get(ctx {{.contextContext|raw}}) {{.clientSetInterface|raw}} {
 	untyped := ctx.Value(Key{})
 	if untyped == nil {
-		{{.loggingFromContext|raw}}(ctx).Panic(
-			"Unable to fetch {{.clientSetInterface}} from context.")
+		if injection.GetConfig(ctx) == nil {
+		    {{.loggingFromContext|raw}}(ctx).Panic(
+		    	    "Unable to fetch {{.clientSetInterface}} from context. This context is not the injection context. Only controllers can inject clients.")
```

Comment:
`"Only controllers can inject clients."` is not quite true, it is constructors called from the sharedmain injection enabled thread or something like that.

I make these calls too from just main threads.

Reply:
mm, ok. "Only the application context (given to controllers via sharedmain)"?

Reply:
given to "constructors", they can be anything, also you can hack it and register a fake informer and grab a client)